### PR TITLE
Docs/125 add the web api of a creating hash api

### PR DESCRIPTION
**Pull request by hitohata**
## Overview of Implementation

Added `generate an invitation link` and `join to a group` API

## Review Points

I put some comments on the code on GitHub.

### CODE REVIEW

File: web-api/nishiki-web-api.yaml [hunk 1]
```diff
@@ -509,6 +545,32 @@ paths:
           description: Deletion complete
         "403":
           description: A user tries to delete a group that is not owned by a requesting user.
+  /groups/{groupId}/?Action=generateInvitaionLink:
```
Comment (hitohata) [inline web-api/nishiki-web-api.yaml:548]: I created a generate invitation link API as a put method.

File: web-api/nishiki-web-api.yaml [hunk 2]
```diff
@@ -445,6 +445,42 @@ paths:
                   groupId:
                     type: string
                     example: d4825cf5-0b1b-dce3-1aa1-ed9adfd6b0bd
+  /groups/?Action=joinToGroup:
```
Comment (hitohata) [inline web-api/nishiki-web-api.yaml:448]: I created a join to a group API as a put method.
The joining user won't know the target group ID, so I decided to not use the {groupId}. This API returns a Group ID that a user joined using this API.

Comment (taniguchiiqqq) [reply web-api/nishiki-web-api.yaml:448]: @hitohata 
Thank you for your description. Let me confirm the behavior. When this endpoint is called, it just returns groupId based on invitationLink, so afterward PUT `/groups/{groupId}/users/{userId}` must be called to add the user to the group. Is this understanding correct?

Comment (hitohata) [reply web-api/nishiki-web-api.yaml:448]: @taniguchiiqqq 

Thank you for your comment.

>so afterward PUT /groups/{groupId}/users/{userId} must be called to add the user to the group. Is this understanding correct?

No, this endpoint will add a user. Therefore, the client (front) doesn't need to more action.

Comment (taniguchiiqqq) [reply web-api/nishiki-web-api.yaml:448]: @hitohata 
I got it. Thanks. Then do you think `PUT /groups/{groupId}/users/{userId}` is neccesarry? When a user creates a group, a user can be added to the group just by calling `POST /groups/{groupId}` as the join to a group endpoint does.

Or both when a user creates a group and when a user joins a group, `PUT /groups/{groupId}/users/{userId}` should be called after getting groupId.

Comment (hitohata) [reply web-api/nishiki-web-api.yaml:448]: @taniguchiiqqq 
It made sense. So, I need to remove the  `PUT /group/{groupId}/users/{userId}` API.

Comment (taniguchiiqqq) [reply web-api/nishiki-web-api.yaml:448]: @hitohata 
> I need to remove the PUT /group/{groupId}/users/{userId} API.

Yes, I agree with this.

Comment (hitohata) [reply web-api/nishiki-web-api.yaml:448]: I deleted it.

**Review by taniguchiiqqq — APPROVED** [review]
Thank you! LGTM

**Review by kotaaaa — APPROVED** [review]
Thank you for creating this doc!!

**Review by uskayyyyy — APPROVED** [review]
LGTM :) Thank you!

**Pull request merged**